js/packages/teams-ai/src/Application.ts

```diff
@@ -148,12 +155,17 @@ export type ConversationUpdateEvents =
  */
 export type RouteHandler<TState extends TurnState> = (context: TurnContext, state: TState) => Promise<void>;
 
+/**
+ * A selector function for matching incoming activities.
+ */
+export type Selector = (context: TurnContext) => Promise<boolean>;
+
 /**
  * Function for selecting whether a route handler should be triggered.
  * @param context Context for the current turn of conversation with the user.
  * @returns A promise that resolves with a boolean indicating whether the route handler should be triggered.
  */
-export type RouteSelector = (context: TurnContext) => Promise<boolean>;
+export type RouteSelector = Selector;
```

Comment:
Why the refactor?

Reply:
I'm using `Selector` type to allow users to "select" the activities in which auto sign in should happen. I didn't want to use `RouteSelector` type because it's not selecting a route.

Reply:
But Selector is the same as what RouteSelector used to be, so how is this different?

Reply:
The difference is when users hover over the property they'll see `Selector` instead of `RouteSelector` in the tooltip documentation. This is to avoid associating the a generic `Selector` with the specific-purpose `RouteSelector` which represents functions for _selecting_ which _route_ to execute. With a generic `Selector` you're simply _selecting_ certain activities, and the added context comes from where it's used. In this case its used as a type for the `autoSignIn` property in the `AuthenticationOptions` interace.

Reply:
I would say distinguishing between `RouteSelector` and `ActivitySelector` (instead of 'just' `Selector` is valuable here. But if you think it's super obvious and I'm just dumb, then feel free to resolve this thread :)

Reply:
Resolving for now - I will ruminate on our 1:1 call and if I think of a different way to distinguish Selector, we can discuss making an update then :)